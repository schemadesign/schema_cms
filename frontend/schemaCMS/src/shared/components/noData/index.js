export { NoData } from './noData.component';
