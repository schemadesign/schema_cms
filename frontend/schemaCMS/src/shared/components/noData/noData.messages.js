/* eslint-disable max-len */
import { defineMessages } from 'react-intl';

export default defineMessages({
  noData: {
    id: 'shared.components.noData.noData',
    defaultMessage: 'No Data Found.',
  },
});
