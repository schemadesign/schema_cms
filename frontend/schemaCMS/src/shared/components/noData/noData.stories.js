import React from 'react';
import { storiesOf } from '@storybook/react';

import { NoData } from './noDadta.component';

storiesOf('Shared Components|NoData', module).add('Default', () => <NoData />);
