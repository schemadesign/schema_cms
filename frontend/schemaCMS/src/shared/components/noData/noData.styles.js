import styled from 'styled-components';

export const NoDataContainer = styled.div`
  color: ${({ theme: { text } }) => text};
  text-align: center;
  padding: 50px 10px;
  font-weight: 200;
  font-size: 20px;
`;
