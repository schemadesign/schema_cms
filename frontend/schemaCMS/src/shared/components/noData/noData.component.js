import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import { always } from 'ramda';

import { renderWhenTrueOtherwise } from '../../utils/rendering';
import { NoDataContainer } from './noData.styles';
import messages from './noData.messages';

export class NoData extends PureComponent {
  static propTypes = {
    children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  };

  renderContent = children =>
    renderWhenTrueOtherwise(always(children), always(<FormattedMessage {...messages.noData} />))(!!children);

  render() {
    const { children } = this.props;

    return <NoDataContainer>{this.renderContent(children)}</NoDataContainer>;
  }
}
