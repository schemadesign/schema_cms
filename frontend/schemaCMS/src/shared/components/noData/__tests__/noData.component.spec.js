import React from 'react';
import { shallow } from 'enzyme';

import { NoData } from '../noData.component';

describe('NoData: Component', () => {
  const defaultProps = {};

  const component = props => <NoData {...defaultProps} {...props} />;

  const render = (props = {}) => shallow(component(props));

  it('should render correctly', () => {
    const wrapper = render();
    global.expect(wrapper).toMatchSnapshot();
  });
});
