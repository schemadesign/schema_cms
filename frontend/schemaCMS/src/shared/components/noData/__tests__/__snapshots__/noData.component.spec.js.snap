// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoData: Component should render correctly 1`] = `
<styled.div>
  <MemoizedFormattedMessage
    defaultMessage="No Data Found."
    id="shared.components.noData.noData"
  />
</styled.div>
`;
